### PR TITLE
Scc 3241 Separate requestability and availability

### PR DESCRIPTION
**Pull request by charmingduchess**
Created two new functions, fixitemrequestability and fixitemavailability to replace old fixitemavailabilityinrespose function.

- also helper functions for edd and phys requestability to separate logic into clear blocks
- replaced requestablebasedonholdingstatusandlocation with requestablebasedonholdinglocation

### CODE REVIEW

File: lib/availability_resolver.js [hunk 1]
```diff
@@ -102,76 +103,71 @@ class AvailabilityResolver {
...
+    }
+  }
 
-        // If it's in ReCAP
+  _fixPhysRequestability (item, isInRecap) {
+    // this is not the final physRequestable value. True physRequestable is
+    // only determined on the front end via the vars closed_locations and
+    // open_locations.
+    if (isInRecap) {
+      return DeliveryLocationsResolver.__deliveryLocationsByCustomerCode.length > 0
```
Comment (nonword) [inline lib/availability_resolver.js:147]: This is a function but you're calling `.length` on it without calling it?

Comment (charmingduchess) [reply lib/availability_resolver.js]: updated to invoke the function

File: lib/availability_resolver.js [hunk 2]
```diff
@@ -102,76 +103,71 @@ class AvailabilityResolver {
...
-        item.physRequestable = false
         item.eddRequestable = false
+        item.physRequestable = false
         item.specRequestable = false
+        item.requestable = [false]
+        const isInRecap = this._isInRecap(item)
+        item.eddRequestable = this._fixEddRequestability(item, isInRecap, request)
+        item.physRequestable = this._fixPhysRequestability(item, isInRecap)
+        item.specRequestable = !!item.aeonUrl
+        if (item.eddRequestable || item.physRequestable || item.specRequestable) {
```
[the remark this answers is not in the document]
Comment (charmingduchess) [reply lib/availability_resolver.js]: updated

**Review by nonword — APPROVED** [review]
Will need a review on QA to check all of the requestability scenarios, but looks good

**Pull request merged**